wren-ai-service/src/web/v1/services/sql_regeneration.py

```diff
@@ -0,0 +1,150 @@
+import logging
+from typing import List, Literal, Optional
+
+from haystack import Pipeline
+from pydantic import BaseModel
+
+from src.utils import async_timer
+
+logger = logging.getLogger("wren-ai-service")
+
+
+# POST /v1/sql-regenerations
+class DecisionPoint(BaseModel):
+    type: Literal["filter", "selectItems", "relation", "groupByKeys", "sortings"]
+    value: str
+
+
+class CorrectionPoint(BaseModel):
+    type: Literal["sql_expression", "nl_expression"]
```

Comment:
The 'nl' probably stands for natural language, but I’m not sure if it’s easy to understand. Perhaps we could add a comment beside it to clarify.